clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy

```diff
@@ -100,6 +101,50 @@ class DeployKubernetesAtomicOperation implements AtomicOperation<DeploymentResul
       replicationControllerBuilder = replicationControllerBuilder.addNewImagePullSecret(imagePullSecret)
     }
 
+
+    if (description.volumeSources) {
+      task.updateStatus BASE_PHASE, "Adding pod volume sources... "
+      def volumeSources = description.volumeSources.collect( { volumeSource ->
+        Volume volume = new Volume(name: volumeSource.name)
+
+        switch (volumeSource.type) {
+          case KubernetesVolumeSourceType.EMPTYDIR:
+            def res = new EmptyDirVolumeSourceBuilder()
+
+            switch (volumeSource.emptyDir.medium) {
+              case KubernetesStorageMediumType.MEMORY:
+                res = res.withMedium("Memory")
+                break
+
+              default:
+                res = res.withMedium("") // Empty string is default...
+            }
+
+            volume.emptyDir = res.build()
+            break
+
+          case KubernetesVolumeSourceType.HOSTPATH:
+            def res = new HostPathVolumeSourceBuilder().withPath(volume.hostPath.path)
+            volume.hostPath = res.build()
+            break
+
+          case KubernetesVolumeSourceType.PERSISTENTVOLUMECLAIM:
+            def res = new PersistentVolumeClaimVolumeSourceBuilder()
+                .withClaimName(volumeSource.persistentVolumeClaim.claimName)
+                .withReadOnly(volumeSource.persistentVolumeClaim.readOnly)
+            volume.persistentVolumeClaim = res.build()
+            break
+
+          default:
+            return null
+        }
+
+        return volume
+      } ).findAll { it }
```

Comment:
I think you can replace `.collect()` with `.findResults{}` and drop the `.findAll{}` altogether.